pkg/apis/build/v1beta1/build_conversion.go

```diff
@@ -0,0 +1,381 @@
+var _ webhook.Conversion = (*Build)(nil)
+
+func (src *Build) ConvertTo(ctx context.Context, obj *unstructured.Unstructured) error {
+	var bs v1alpha1.Build
+
+	bs.TypeMeta = src.TypeMeta
+	bs.TypeMeta.APIVersion = alphaGroupVersion
+
+	bs.ObjectMeta = src.ObjectMeta
+
+	src.Spec.ConvertTo(&bs.Spec)
+	mapito, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&bs)
+	if err != nil {
+		ctxlog.Error(ctx, err, "failed structuring the newObject")
+	}
+	obj.Object = mapito
+
+	return nil
+
+}
+
+func (src *Build) ConvertFrom(ctx context.Context, obj *unstructured.Unstructured) error {
+
+	var bs v1alpha1.Build
+
+	unstructured := obj.UnstructuredContent()
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructured, &bs)
+	if err != nil {
+		ctxlog.Error(ctx, err, "failed unstructuring the convertedObject")
+	}
+	src.ObjectMeta = bs.ObjectMeta
+	src.TypeMeta = bs.TypeMeta
+	src.TypeMeta.APIVersion = betaGroupVersion
+
+	src.Spec.ConvertFrom(&bs.Spec)
+
+	src.Status = BuildStatus{
+		Registered: bs.Status.Registered,
+		Reason:     (*BuildReason)(bs.Status.Reason),
+		Message:    bs.Status.Message,
+	}
+
+	return nil
+}
+
+func (dest *BuildSpec) ConvertFrom(orig *v1alpha1.BuildSpec) error {
+	// Handle BuildSpec Source
+	specSource := Source{}
+	if orig.Source.BundleContainer != nil {
+		specSource.Type = OCIArtifactType
+		specSource.OCIArtifact = &OCIArtifact{
+			Image:      orig.Source.BundleContainer.Image,
+			Prune:      (*PruneOption)(orig.Source.BundleContainer.Prune),
+			PullSecret: &orig.Source.Credentials.Name,
+		}
+	} else {
+		specSource.Type = GitType
+		specSource.GitSource = &Git{
+			URL:      orig.Source.URL,
+			Revision: orig.Source.Revision,
+		}
+		if orig.Source.Credentials != nil {
+			specSource.GitSource.CloneSecret = &orig.Source.Credentials.Name
+		}
+	}
+	specSource.ContextDir = orig.Source.ContextDir
+	dest.Source = specSource
+
+	// Handle BuildSpec Triggers
+	if orig.Trigger != nil {
+		dest.Trigger = &Trigger{}
+		for i := range orig.Trigger.When {
+			dest.Trigger.When = append(dest.Trigger.When, convertToBetaTriggers(&orig.Trigger.When[i]))
+		}
+		if orig.Trigger.SecretRef != nil {
+			dest.Trigger.TriggerSecret = &orig.Trigger.SecretRef.Name
+		}
+	}
+
+	// Handle BuildSpec Strategy
+	dest.Strategy = Strategy{
+		Name:       orig.StrategyName(),
+		Kind:       (*BuildStrategyKind)(orig.Strategy.Kind),
+		APIVersion: orig.Strategy.APIVersion,
+	}
+
+	// Handle BuildSpec ParamValues
+	dest.ParamValues = []ParamValue{}
+	for _, p := range orig.ParamValues {
+		param := convertBetaParamValue(p)
+		dest.ParamValues = append(dest.ParamValues, param)
+	}
+
+	//handle spec.Dockerfile migration
+	if orig.Dockerfile != nil {
+		dockerfileParam := ParamValue{
+			Name: "PARAM_DOCKERFILE",
+			SingleValue: &SingleValue{
+				Value: orig.Dockerfile,
+			},
+		}
+		dest.ParamValues = append(dest.ParamValues, dockerfileParam)
+	}
+
+	// Handle BuildSpec Output
+	dest.Output.Image = orig.Output.Image
```

Comment:
Output.Insecure is missing

Reply:
so we migrate this into a parameter? by which name?

Reply:
No, the insecure flag should exist in both alpha and beta under output. Seems like it was forgotten.